### ISSUE TITLE
Feature Request: Best effort reading of game files even if storage device information is incomplete (Windows)

### ISSUE THREAD

**Issue by allisaurus** [o3de/o3de#16418]
**Is your feature request related to a problem? Please describe.**
While experimenting with running a Windows build of [O3DE Multiplayer Sample](https://github.com/o3de/o3de-multiplayersample) on [WINE](https://github.com/ValveSoftware/wine), ran into issues getting the server executable to work. Looking into the logs revealed this:
```
Launcher: The asset cache folder of [DRIVE_PATH]\Cache\pc has been successfully read from the Settings Registry
SettingsRegistryMergeUtils: Optional config file "[DRIVE_PATH]\Cache\pc\user.cfg" not found.
Streamer: Skipping drive 'C:\' because device does not support queuing requests.
Streamer: Skipping drive [DRIVE_NAME]' because device does not support queuing requests.
```

Digging into the code around that error message, It looks like [O3DE is calling the `DeviceIOControl` WinAPI](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Platform/Windows/AzCore/IO/Streamer/StreamerConfiguration_Windows.cpp#L37-L94) to determine whether a drive supports [_CommandQueueing_](https://learn.microsoft.com/en-us/windows/win32/api/winioctl/ns-winioctl-storage_adapter_descriptor) (among other things), and if the drive returns `false`, as [WINE is currently hardcoded to do](https://github.com/ValveSoftware/wine/blob/cd165953c8b379a78418711f07417022e503c81b/dlls/mountmgr.sys/device.c#L1571), then O3DE [skips the drive](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Platform/Windows/AzCore/IO/Streamer/StreamerConfiguration_Windows.cpp#L328-L335) altogether. This results in the game executable unable to load any of its assets from that drive: 
```
Spawnables: An error occurred during the loading of spawnable 'levels/newstarbase/newstarbase.spawnable' ({67155E6B-0365-5033-8A95-C193C1334F5B}:cd45fd06).
```

I'm not familiar enough with the file streaming portion of O3DE to know how detrimental attempting to read files from a drive we can't fully describe would be, or how frequently we'd run into this kind of scenario outside of WINE, however it feels weird that the game executable can't read files from the same drive it's running from.

**Describe the solution you'd like**
Attempt to read game files even when the underlying drive fails to report specific information. Perhaps by falling back to sane defaults re: file streaming so that a game/project can potentially still run?

**Describe alternatives you've considered**
N/A

**Comment by AMZN-Olex** [comment]
Is this with a Server Launcher or Game Launcher?

**Comment by allisaurus** [comment]
@AMZN-Olex graphics on. Specific execution command was `MultiplayerSample.ServerLauncher.exe --console-command-file=launch_server.cfg -bg_ConnectToAssetProcessor=0 -sys_PakPriority=2 -sv_terminateOnPlayerExit=false`

**Comment by lumberyard-employee-dm** [comment]
The Streamer Configuration can be updated to use the Generic Streamer profile instead of the Windows specific streamer profile.
I believe we can override the drive settings in the [Registry/Platform/Windows/streamer.launcher.profile.setreg](https://github.com/o3de/o3de/blob/development/Registry/Platform/Windows/streamer.launcher.profile.setreg#L26) file to not include the `AZ::IO::WindowsStorageDriveConfig`.

That would lead to use the generic Streamer configuration, that doesn't use the WinAPI specific calls

**Comment by cgalvan** [comment]
I haven't confirmed when running on WINE yet, but I did confirm that removing the specific windows drive config results in the generic `AZ::IO::StorageDriveConfig` being used instead, which does work as expected when running the MultiplayerSample unified launcher.

**Comment by allisaurus** [comment]
I've confirmed that after enforcing the generic `AZ::IO::StorageDriveConfig` w/ setreg values, O3DE MPS does run on my WINE setup 😎 

This validates the workaround, but doesn't change the default file streaming behavior when underlying device details aren't recognized. Is there a further fix here to be made, or is this kind of configuration expected to be necessary when running O3DE outside a specific set of supported environments?

**Comment by lumberyard-employee-dm** [comment]
The further fix here would be to add some kind of detection to determine if we are either running under WINE or running on a legitimate Windows OS.
With that fix, we can add a Settings Registry specialization, to to filter either WINE specific setreg files (such as `"streamer.launcher.profile.wine.setreg"` or "super" official windows specific setreg files (such as `stramer.launcher.profile.IAmReallyRunningOnWindows.setreg"`).

**Comment by lumberyard-employee-dm** [comment]
Or maybe a proper solution is to allow the "Windows" drive to be used even if command queuing is not enabled in the Streamer code itself https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Platform/Windows/AzCore/IO/Streamer/StreamerConfiguration_Windows.cpp#L306-L335

**Comment by cgalvan** [comment]
```
bool CollectIoHardwareInformation(HardwareInformation& info, bool includeAllHardware, bool reportHardware)
    {
        if (!CollectHardwareInfo(info, includeAllHardware, reportHardware))
        {
            // The numbers below are based on common defaults from a local hardware survey.
            info.m_maxPageSize = 4096;
            info.m_maxTransfer = 512_kib;
            info.m_maxPhysicalSectorSize = 4096;
            info.m_maxLogicalSectorSize = 512;
            info.m_profile = "Generic";
        }
        return true;
    }
```

If we fail to find any usable drives, should we fallback to trying with the `AZ::IO::StorageDriveConfig`?

**Comment by lumberyard-employee-dm** [comment]
We shouldn't need to, as long as `CollectHardwareInfo` returns false.

**Comment by cgalvan** [comment]
Good point, I can investigate on my Ubuntu machine why it isn't returning false when running on Wine.
